### PR TITLE
Clean up the code a bit

### PR DESCRIPTION
**Pull request by oponder**
Adds tests that run without an internet connection.
Will refactor the code to try and formalise the parameter checking more.

Some small fixes along the way:

instanceLogs method says it can take 'null' for numLines, but that raised an error [fixed]

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -182,17 +194,15 @@ var GiantSwarm = (function () {
       if (typeof successCallback !== 'function') throw new this.UserException("Parameter successCallback must be of type function");
       if (errorCallback !== null && typeof errorCallback !== 'function') throw new this.UserException('errorCallback must be null or function');
       params = params || '';
-      var headers = {
-        'Authorization': 'giantswarm ' + authToken
-      }
-      if (clusterId) {
-        headers[CLUSTER_ID_HEADER] = clusterId;
-      }
+      var headers = { 'Authorization': 'giantswarm ' + authToken }
```
Comment (ZeissS) [inline lib/client.js]: did the price of newlines rise?

File: lib/client.js [hunk 2]
```diff
@@ -466,90 +477,81 @@ var GiantSwarm = (function () {
...
+      this._validateStringParameters({
+        organizationName: organizationName,
+        environmentName: environmentName,
+        serviceName: serviceName,
+        componentName: componentName
+      });
+
+      params = JSON.stringify({component: componentName})
+
+      var uri = '/v1/org/' + encodeURIComponent(organizationName) + '/env/' + encodeURIComponent(environmentName);
```
Comment (ZeissS) [inline lib/client.js]: these lines repeat them self a few times - maybe move to a separate helper function?

File: lib/client.js (hunk 1 above)
Comment (oponder) [inline lib/client.js]: I feel like spreading this out on multiple lines doesn't help my readability.
The single line is more readable to me. What do you think?

File: lib/client.js (hunk 2 above)
Comment (oponder) [inline lib/client.js]: yup! thx

File: lib/client.js (hunk 1 above)
Comment (marians) [inline lib/client.js]: I don't mind the newlines, but I'm missing a semicolon.

**Comment by oponder** [comment]
Todo:
[] Fix streamLogs
[] Fix streamStats
[] eslint

**Comment by oponder** [comment]
@marians 
This is ready for review. There are still eslint issues, but I don't want to hold back this issue because of that.

`jasmine` runs the tests, and `gulp eslint` runs eslint.

```
    1:20  error  "require" is not defined                                no-undef
    2:16  error  "require" is not defined                                no-undef
   18:13  error  "require" is not defined                                no-undef
   21:19  error  "window" is not defined                                 no-undef
   60:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
   90:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  102:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  163:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  230:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  258:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  268:22  error  Unexpected dangling "_" in "_processEnvironmentNames"   no-underscore-dangle
  291:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  300:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  314:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  324:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  338:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  348:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  362:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  374:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  388:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  400:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  415:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  432:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  448:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  465:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  481:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  498:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  510:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  521:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  534:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  556:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  570:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  579:6   error  Unexpected dangling "_" in "_postForWebSocket"          no-underscore-dangle
  598:6   error  Unexpected dangling "_" in "_validateStringParameters"  no-underscore-dangle
  606:6   error  Unexpected dangling "_" in "_postForWebSocket"          no-underscore-dangle
  628:17  error  Unexpected dangling "_" in "_headers"                   no-underscore-dangle
  658:2   error  "window" is not defined                                 no-undef
  661:2   error  "module" is not defined                                 no-undef
```

**Comment by oponder** [comment]
Found a todo: Make sure it works with servicemonitor too. Right now the service monitor fails silently with this branch. Timeline runs, but no stats are showing up. Will take a look

**Comment by oponder** [comment]
Streams work again now too.

**Comment by oponder** [comment]
LGTM via https://github.com/giantswarm/webclient/pull/97

**Pull request merged**